Grade by max-scoring submission.

ncik-roberts opened the pull request:
Allow instructor to choose between grading the max-scoring
submission and the latest submission.

I set the default to scoring by max score, since this is what we want in compilers.

The setting is managed on a per-assessment basis, and can be edited from the normal edit-assessment UI.

Commit:
Add assessment setting for grading mode.

Allow instructor to choose between grading the max-scoring
submission and the latest submission.

Commit:
Set default to grading by max score.

jburns20 commented on app/models/assessment.rb:
```diff
@@ -491,7 +491,8 @@ def active?
                  "has_autograde" => nil,
-                 "has_scoreboard" => nil }
+                 "has_scoreboard" => nil,
+                 "grade_latest" => false }
```
We should keep the default as-is (latest score), since this could potentially be merged back into mainline Autolab in the future.

jburns20 reviewed (changes requested):
Since it isn't clear whether the assessment uses max-score or latest-score grading when looking at the assessment page, we should probably add a line just above the most recent submissions (and at the top of the submission history page) that says which grading scheme is in effect.

The other place we should probably respect the max-submission grading is on the instructor gradesheet page. It would be handy to be able to see everyone's scores at a glance. Updating that might require more work, though.

Commit:
Update latest_submission with graded submission.

Repurpose the latest_submission field in the db to store the id
of the submission that is graded. On editing a submission, this
column is updated.

ncik-roberts commented:
Sorry, had to force push because I messed something up in the history. The third commit is the one I added since the change request.

The strategy I took was repurposing the `latest_submission` column in the assessments table as the id of the _graded_ submission.

jburns20 committed:
Make grade_latest default to true when importing assessments

jburns20 committed:
Rename penalties tab to grading tab
